grudge/execution.py

```diff
@@ -543,6 +543,9 @@ def prg(nmatrices):
             self.array_context.call_loopy(
                     prg(noperators),
                     diff_mat=matrices_ary_dev,
+                    nelements=in_grp.nelements,
+                    nunit_nodes_in=in_grp.nunit_dofs,
+                    nunit_nodes_out=out_grp.nunit_dofs,
```

Comment:
I'm a little confused that this is even necessary. Normally, Loopy would infer these from the array sizes. At least for now though, we fix those parameters, (Ctrl-F `fix_parameters` in https://github.com/inducer/meshmode/pull/116/files).  But `loopy.fix_parameters` [removes](https://github.com/inducer/loopy/blob/09c263f876b230fc46007874b9059476cbdd5a68/loopy/transform/parameter.py#L100-L110) arguments it fixes, so it's not clear to me why these are even still around.

@kaushikcfd, what do you think?

Reply:
I agree that nunit_nodes_(in|out) should be just fixed and not passed as arguments.

Reply:
So, add this as a parameter to `prg()` and call `fix_parameters` in there?

Reply:
Instead of passing `prg(operators)` we can replace it with `lp.fix_parameters(prg(operators), nunit_nodes_in=in_grp.nunit_dofs, nunint_nodes_out=out_grp.nunit_dofs)`.

Reply:
Sounds good. Note that there is still a call to `lp.fix_parameters` inside `prg()`, for `nmatrices`.

Reply:
Yep, I think that's reasonable.